addons/mrp/models/mrp_production.py

```diff
@@ -270,6 +276,29 @@ def _compute_allowed_product_ids(self):
                     product_domain += [('id', 'in', production.bom_id.product_tmpl_id.product_variant_ids.ids)]
             production.allowed_product_ids = self.env['product.product'].search(product_domain)
 
+    @api.depends('move_raw_ids.availability_indication', 'move_raw_ids.availability_state')
+    def _compute_components_availability(self):
+        inactive_orders = self.filtered(lambda order: order.state in ['cancel', 'draft', 'done', 'to_close'])
+        inactive_orders.write({
+            'components_availability': '',
+            'components_availability_state': 'available'
+        })
+        active_orders = (self - inactive_orders)
+        for order in active_orders:
+            worst_moves = order.move_raw_ids.filtered(lambda move: move.availability_state == 'late')
+            worst_moves = worst_moves or order.move_raw_ids.filtered(lambda move: move.availability_state == 'partial')
+            worst_moves = worst_moves or order.move_raw_ids.filtered(lambda move: move.availability_state == 'waiting')
+            if worst_moves:
+                if worst_moves[0].availability_state == 'partial':
+                    order.components_availability_state = 'late'
+                    order.components_availability = 'Not Available'
+                else:
+                    order.components_availability_state = worst_moves[0].availability_state
+                    order.components_availability = worst_moves[0].availability_indication
+            else:
+                order.components_availability_state = 'available'
+                order.components_availability = 'Available'
```

Comment:
Yeah, using `sorted` instead of multiple `filtered` looks pretty damn cool :frog: 
Also, I moved this part into `stock_move.py` like you suggested elsewhere:

```python3
def _get_availability_vals(self):
    vals = ('available', 'Available')
    worst_move = self.sorted(key=lambda move: (
        move.availability_state == 'late',
        move.availability_state == 'partial',
        move.availability_state == 'waiting'
    ))[:1]
    if worst_move:
        if worst_move.availability_state == 'partial':
            vals = ('late', 'Not Available')
        else:
            vals = (
                worst_move.availability_state,
                worst_move.availability_indication,
            )
    return vals
```